bluepyopt/ephys/extra_features_utils.py

```diff
@@ -0,0 +1,500 @@
+    potential between trough and baseline.
+
+    Optionally the function returns also the indices per waveform where the
+    potential crosses baseline.
+
+    Parameters
+    ----------
+    waveforms  : numpy.ndarray (num_waveforms x num_samples)
+        waveforms to compute features for
+    sampling_frequency  : float
+        rate at which the waveforms are sampled (Hz)
+    return_idx : bool
+        if true, also returns index of threshold crossing before and
+        index of threshold crossing after peak
+
+    Returns
+    -------
+
+    np.ndarray or (np.ndarray, np.ndarray)
+        Repolarization slope of the waveforms or (Repolarization slope of the
+        waveforms, return to base index)
+    """
+    trough_idx, peak_idx = _get_trough_and_peak_idx(waveforms)
+
+    rslope = np.empty(waveforms.shape[0])
+    rslope[:] = np.nan
+    return_to_base_idx = np.empty(waveforms.shape[0], dtype=np.int)
+    return_to_base_idx[:] = 0
+
+    time = np.arange(0, waveforms.shape[1]) * (1 / sampling_frequency)  # in s
+    for i in range(waveforms.shape[0]):
+        if trough_idx[i] == 0:
+            continue
+
+        rtrn_idx = np.where(waveforms[i, trough_idx[i]:] >= 0)[0]
+        if len(rtrn_idx) == 0:
+            continue
+
+        return_to_base_idx[i] = (
+            rtrn_idx[0] + trough_idx[i]
+        )  # first time after  trough, where waveform is at baseline
+
+        if return_to_base_idx[i] - trough_idx[i] < 3:
+            continue
+        slope = _get_slope(
+            time[trough_idx[i]:return_to_base_idx[i]],
+            waveforms[i, trough_idx[i]:return_to_base_idx[i]]
+        )
+        rslope[i] = slope[0]
+
+    if not return_idx:
+        return rslope
+    else:
+        return rslope, return_to_base_idx
+
+
+def recovery_slope(waveforms, sampling_frequency, window):
+    """
+    Return the recovery slope of input waveforms. After repolarization,
+    the neuron hyperpolarizes until it peaks. The recovery slope is the
+    slope of the action potential after the peak, returning to the baseline
+    in dV/dT. The slope is computed within a user-defined window after
+    the peak.
+
+    Takes a numpy array of waveforms and returns an array with
+    recovery slopes per waveform.
+
+    Parameters
+    ----------
+    waveforms  : numpy.ndarray (num_waveforms x num_samples)
+        waveforms to compute features for
+    sampling_frequency  : float
+        rate at which the waveforms are sampled (Hz)
+    window : float
+        length after peak wherein to compute recovery slope (ms)
+
+
+    Returns
+    -------
+    np.ndarray
+        Recovery slope of the waveforms
+    """
+    _, peak_idx = _get_trough_and_peak_idx(waveforms)
+    rslope = np.empty(waveforms.shape[0])
+    rslope[:] = np.nan
+
+    time = np.arange(0, waveforms.shape[1]) * (1 / sampling_frequency)  # in s
+
+    for i in range(waveforms.shape[0]):
+        if peak_idx[i] == 0 or peak_idx[i] == waveforms.shape[1]:
```

Comment:
this can be reduced to a single comparison with the `in` operator
 
`if peak_idx[i] in [0, waveforms.shape[1]]:`